server/app/services/applicant/ApplicantService.java

```diff
@@ -851,6 +852,31 @@ public CompletionStage<ApplicationPrograms> relevantProgramsForApplicant(long ap
             httpExecutionContext.current());
   }
 
+  /**
+   * Find programs the applicant may be eligible for.
+   *
+   * @return All programs that are appropriate to serve to an applicant and that they may be
+   *     eligible for. Includes programs with matching eligibility criteria or no eligibility
+   *     criteria.
+   *     <p>Does not include the Common Intake Form.
+   *     <p>"Appropriate programs" those returned by {@link #relevantProgramsForApplicant(long)}.
+   */
+  public CompletionStage<ImmutableList<ApplicantProgramData>> maybeEligibleProgramsForApplicant(
+      long applicantId) {
+    return relevantProgramsForApplicant(applicantId)
+        .thenApplyAsync(
```

Comment:
**suggestion (blocking):** pass the http execution context as a second argument to `thenApplyAsync` https://www.playframework.com/documentation/2.8.x/ThreadPools#Using-the-default-thread-pool

Reply:
Done